src/audio/pipeline/pipeline-graph.c

```diff
@@ -431,6 +431,11 @@ struct comp_dev *pipeline_get_dai_comp(uint32_t pipeline_id, uint32_t *latency)
 		buffer = buffer_from_list(comp_buffer_list(sink->cd, PPL_DIR_DOWNSTREAM)->next,
 					  struct comp_buffer, PPL_DIR_DOWNSTREAM);
 		buff_comp = buffer_get_comp(buffer, PPL_DIR_DOWNSTREAM);
+
+		/* buffer_comp is in another incomplete pipeline */
+		if (!buff_comp->pipeline)
```

Comment:
Isn't this protecting a state change whilst another connected pipeline is being built ?

Reply:
what is the error that you see @RanderWang? this function seems to be called during trigger. Why would buffer->comp->pipeline ever be NULL at that point?

Reply:
> But if this would have ever happened we would have a NULL pointer dereference in calling ipc_get_ppl_sink_comp() at line 439.

sure. but it will first fw panic at here

Reply:
> Isn't this protecting a state change whilst another connected pipeline is being built ?

NO, other pipelines will be built after this trigger of pipeline. 
The sequence of windows is : (assume there are two pipelines A and B in the stream)
(1) pipeline A & B create 
(2) create and bind module in pipeline A & B 
(3) pipeline A & B reset  ( at this time, FW know pipeline is complete and set A & B to complete)
(4) pipeline A & B pause
(5) pipeline A & B start

For Linux driver
(1) pipeline A & B create 
(2) create and bind module in pipeline A & B 
(4) pipeline A pause ( at this time, FW know pipeline is complete and set A to complete)
(5) pipeline A trigger start  (B is not complete, so you can't access pipeline in component in pipeline B)
(6) pipeline B pause ( at this time, FW know pipeline is complete and set B to complete)
(7) pipeline B trigger start  (A is complete)